modules/andes-core/client/src/main/java/org/wso2/andes/client/XAConnectionImpl.java

```diff
@@ -145,9 +178,35 @@ synchronized void deregisterSession(XASession_9_1 xaSession) {
      *
      * @throws JMSException if there was an error closing the physical connection
      */
-    synchronized void internalClose() throws JMSException {
+    void internalClose() throws JMSException {
+        boolean closeSuccessful = closeIfNoActiveSessions();
+        if (closeSuccessful) {
```

Comment:
Do we need to assign it to closeSuccessful ? Cant we use the closeIfNoActiveSessions() itself as the condition ?